2023-09-18 17:46:26.207980: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:46:26.208152: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:46:33,757 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:46:33,828 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:46:33,828 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:46:33,828 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:46:34,026 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:46:34,027 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,043 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.44}
INFO flwr 2023-09-18 17:46:34,043 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:46:34,141 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:46:34,163 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:46:34,165 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,173 | server.py:125 | fit progress: (1, 0.457846118870447, {'accuracy': 0.7933333333333333}, 0.12964619999999982)
DEBUG flwr 2023-09-18 17:46:34,174 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:46:34,195 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 17:46:34,195 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:46:34,195 | server.py:223 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:46:34,212 | server.py:237 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,223 | server.py:125 | fit progress: (2, 0.4463443621068283, {'accuracy': 0.8}, 0.17976559999999964)
DEBUG flwr 2023-09-18 17:46:34,223 | server.py:173 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:46:34,246 | server.py:187 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:46:34,246 | server.py:223 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:46:34,274 | server.py:237 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,285 | server.py:125 | fit progress: (3, 0.4472113037058698, {'accuracy': 0.7933333333333333}, 0.24169310000000088)
DEBUG flwr 2023-09-18 17:46:34,285 | server.py:173 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:46:34,304 | server.py:187 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:46:34,304 | server.py:223 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:46:34,335 | server.py:237 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,346 | server.py:125 | fit progress: (4, 0.4472042571436111, {'accuracy': 0.7933333333333333}, 0.3030495000000002)
DEBUG flwr 2023-09-18 17:46:34,347 | server.py:173 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:46:34,368 | server.py:187 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:46:34,368 | server.py:223 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:46:34,396 | server.py:237 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,407 | server.py:125 | fit progress: (5, 0.4472035937221491, {'accuracy': 0.7933333333333333}, 0.36332390000000103)
DEBUG flwr 2023-09-18 17:46:34,407 | server.py:173 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:46:34,430 | server.py:187 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:46:34,431 | server.py:223 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:46:34,461 | server.py:237 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,473 | server.py:125 | fit progress: (6, 0.44503511623772835, {'accuracy': 0.7966666666666666}, 0.4299023999999996)
DEBUG flwr 2023-09-18 17:46:34,473 | server.py:173 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:46:34,495 | server.py:187 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:46:34,495 | server.py:223 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:46:34,526 | server.py:237 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,538 | server.py:125 | fit progress: (7, 0.44505494277050955, {'accuracy': 0.7966666666666666}, 0.49438830000000067)
DEBUG flwr 2023-09-18 17:46:34,538 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:46:34,571 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:46:34,571 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:46:34,605 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,617 | server.py:125 | fit progress: (8, 0.44388600518509586, {'accuracy': 0.8}, 0.5736027000000004)
DEBUG flwr 2023-09-18 17:46:34,617 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:46:34,647 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:46:34,647 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:46:34,680 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,693 | server.py:125 | fit progress: (9, 0.4438670269395082, {'accuracy': 0.8}, 0.6498860000000004)
DEBUG flwr 2023-09-18 17:46:34,693 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:46:34,722 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:46:34,723 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:46:34,759 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:46:34,771 | server.py:125 | fit progress: (10, 0.44386756248049913, {'accuracy': 0.8}, 0.7278976999999998)
DEBUG flwr 2023-09-18 17:46:34,771 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:46:34,796 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:46:34,796 | server.py:147 | FL finished in 0.7530584999999999
INFO flwr 2023-09-18 17:46:34,797 | app.py:218 | app_fit: losses_distributed [(1, 0.3935479484498501), (2, 0.3995800673961639), (3, 0.3991947412490845), (4, 0.3991946150859197), (5, 0.398563367270288), (6, 0.3976429566031411), (7, 0.3998355750526701), (8, 0.3995296380349568), (9, 0.39952928168433055), (10, 0.399529259119715)]
INFO flwr 2023-09-18 17:46:34,797 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:46:34,797 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:46:34,797 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.457846118870447), (2, 0.4463443621068283), (3, 0.4472113037058698), (4, 0.4472042571436111), (5, 0.4472035937221491), (6, 0.44503511623772835), (7, 0.44505494277050955), (8, 0.44388600518509586), (9, 0.4438670269395082), (10, 0.44386756248049913)]
INFO flwr 2023-09-18 17:46:34,797 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.44), (1, 0.7933333333333333), (2, 0.8), (3, 0.7933333333333333), (4, 0.7933333333333333), (5, 0.7933333333333333), (6, 0.7966666666666666), (7, 0.7966666666666666), (8, 0.8), (9, 0.8), (10, 0.8)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.11425105,  1.0897684 ,  0.84627421,  0.42304894,  0.38249968,
         0.11596161, -0.12373099,  0.01973364]]), array([-0.06462968])]
[array([[-2.11425105,  1.0897684 ,  0.84627421,  0.42304894,  0.38249968,
         0.11596161, -0.12373099,  0.01973364]]), array([-0.06462968])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.06675289,  1.05168569,  0.80377142,  0.45920602,  0.31092005,
         0.14364991, -0.04185474,  0.00343026]]), array([-0.12811579])]
[array([[-2.06675289,  1.05168569,  0.80377142,  0.45920602,  0.31092005,
         0.14364991, -0.04185474,  0.00343026]]), array([-0.12811579])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01075782,  0.9807542 ,  0.79872453,  0.42524628,  0.29193665,
         0.12820522, -0.03539109,  0.0086337 ]]), array([-0.00330019])]
[array([[-2.01075782,  0.9807542 ,  0.79872453,  0.42524628,  0.29193665,
         0.12820522, -0.03539109,  0.0086337 ]]), array([-0.00330019])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01093446,  0.98059107,  0.79883119,  0.42518706,  0.29199064,
         0.12873159, -0.03551459,  0.00842536]]), array([-0.00318984])]
[array([[-2.01093446,  0.98059107,  0.79883119,  0.42518706,  0.29199064,
         0.12873159, -0.03551459,  0.00842536]]), array([-0.00318984])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01087284,  0.98056323,  0.79881971,  0.42517207,  0.29198999,
         0.12873391, -0.03550884,  0.00842611]]), array([-0.0032129])]
[array([[-2.01087284,  0.98056323,  0.79881971,  0.42517207,  0.29198999,
         0.12873391, -0.03550884,  0.00842611]]), array([-0.0032129])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.00191989e+00,  1.04117452e+00,  8.46348325e-01,
         4.38242460e-01,  2.97464561e-01,  1.24998654e-01,
         8.21438221e-04, -8.76745648e-03]]), array([-0.21369711])]
[array([[-2.00191989e+00,  1.04117452e+00,  8.46348325e-01,
         4.38242460e-01,  2.97464561e-01,  1.24998654e-01,
         8.21438221e-04, -8.76745648e-03]]), array([-0.21369711])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.00188114e+00,  1.04126996e+00,  8.46364662e-01,
         4.38107500e-01,  2.97453986e-01,  1.24365926e-01,
         9.97649021e-04, -8.70615415e-03]]), array([-0.21327326])]
[array([[-2.00188114e+00,  1.04126996e+00,  8.46364662e-01,
         4.38107500e-01,  2.97453986e-01,  1.24365926e-01,
         9.97649021e-04, -8.70615415e-03]]), array([-0.21327326])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03694774,  1.02301612,  0.81072007,  0.42486604,  0.28498366,
         0.12078402,  0.02591459,  0.01458118]]), array([-0.14124351])]
[array([[-2.03694774,  1.02301612,  0.81072007,  0.42486604,  0.28498366,
         0.12078402,  0.02591459,  0.01458118]]), array([-0.14124351])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.037258  ,  1.02294225,  0.81080252,  0.4251418 ,  0.28489988,
         0.12094127,  0.02601285,  0.0146188 ]]), array([-0.14159964])]
[array([[-2.037258  ,  1.02294225,  0.81080252,  0.4251418 ,  0.28489988,
         0.12094127,  0.02601285,  0.0146188 ]]), array([-0.14159964])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03727355,  1.02295117,  0.81081121,  0.42515674,  0.28489825,
         0.12094426,  0.0259946 ,  0.01462948]]), array([-0.1416156])]
[array([[-2.03727355,  1.02295117,  0.81081121,  0.42515674,  0.28489825,
         0.12094426,  0.0259946 ,  0.01462948]]), array([-0.1416156])]
>>>>>>>>>>>>>run time:
0.484375s
